src/lightning/fabric/strategies/deepspeed.py

```diff
@@ -477,7 +481,7 @@ def load_checkpoint(
             tag="checkpoint",
             load_optimizer_states=optimzer_state_requested,
             load_lr_scheduler_states=False,
-            load_module_strict=True,  # TODO(fabric): make strict loading configurable
+            load_module_strict=strict,
```

Comment:
This applies only to the module keys here. Below in the for-loop we also handle the client state (remaining user state dict) and strictness should also be checked there too I think. The same applies to the fsdp strategy. below. We can leave it as a follow-up for a different PR if you prefer.